webapp/tests/test_render_glyph.py

```diff
@@ -0,0 +1,1002 @@
+
+    def test_safeSum_empty_string(self):
+      self.assertEqual(glyph.safeSum(''), 0)
+
+    def test_safeSum_list_with_string(self):
+      with self.assertRaises(TypeError):
+        glyph.safeSum(['a',10,30])
+
+    def test_safeSum_list_all_numbers(self):
+      self.assertEqual(glyph.safeSum([1,5.1,6,]), 12.1)
+
+    def test_safeSum_list_with_nan(self):
+      self.assertEqual(glyph.safeSum([1,5.1,6,float('Nan')]), 12.1)
+
+    #
+    # Testing any()
+    #
+    def test_any_None(self):
+      with self.assertRaises(TypeError):
+        glyph.any(None)
+
+    def test_any_number_not_list(self):
+      with self.assertRaises(TypeError):
+        glyph.any(1)
+
+    def test_any_empty_string(self):
+      self.assertFalse(glyph.any(''))
+
+    def test_any_list_with_false(self):
+      self.assertFalse(glyph.any([False]))
+
+    def test_any_list_with_0(self):
+      self.assertFalse(glyph.any([0]))
+
+    def test_any_list_with_string(self):
+      self.assertTrue(glyph.any(['a',10,30]))
+
+    def test_any_list_all_numbers(self):
+      self.assertTrue(glyph.any([1,5.1,6,]))
+
+    def test_any_list_with_nan(self):
+      self.assertTrue(glyph.any([1,5.1,6,float('Nan')]), 12.1)
+
+    #
+    # Testing dataLimits
+    #
+    def test_dataLimits_defaults(self):
+        seriesList = self._generate_series_list()
+        self.assertEqual(glyph.dataLimits(seriesList), (0,100))
+
+    def test_dataLimits_defaults_ymin_positive(self):
+        config = [range(10, 101), range(10, 101), [1, None, None, None, None]]
+        seriesList = self._generate_series_list(config)
+        self.assertEqual(glyph.dataLimits(seriesList), (1,100))
+
+    def test_dataLimits_empty_list(self):
+        self.assertEqual(glyph.dataLimits([]), (0.0, 1.0))
+
+    def test_dataLimits_drawNull(self):
+        seriesList = self._generate_series_list()
+        self.assertEqual(glyph.dataLimits(seriesList, True, False), (0,100))
+
+    def test_dataLimits_stacked(self):
+        seriesList = self._generate_series_list()
+        self.assertEqual(glyph.dataLimits(seriesList, False, True), (0, 8))
+
+    def test_dataLimits_drawNull_stacked(self):
+        seriesList = self._generate_series_list()
+        self.assertEqual(glyph.dataLimits(seriesList, True, True), (0, 8))
+
+    def test_dataLimits_drawNull_stacked_no_missing(self):
+        config = [range(10, 101), range(10, 101), range(100,300)]
+        seriesList = self._generate_series_list(config)
+        self.assertEqual(glyph.dataLimits(seriesList, True, True), (10, 390))
+
+    def test_dataLimits_drawNull_ymin_positive_missing_data(self):
+        config = [range(10, 101), range(10, 101), [1, None, None, None, None]]
+        seriesList = self._generate_series_list(config)
+        self.assertEqual(glyph.dataLimits(seriesList, True, False), (0.0, 100))
+
+    def test_dataLimits_drawNull_ymax_negative_missing_data(self):
+        config = [range(-10, -101), range(-10, -101), [-50, None, None, None, None]]
```

Comment:
`range(-10, -101)` is `[]` (or the iterator equivalent in Python 3+). If that's what you want, it's probably clearer to write `[]` here. Alternatively, maybe you meant to add a third parameter?: `range(-10, -101, -1)`.